db/version_builder.cc

```diff
@@ -341,28 +339,26 @@ class VersionBuilder::Rep {
 #ifndef NDEBUG
       FileMetaData* prev_file = nullptr;
 #endif
-
-      for (const auto& added : added_files) {
+      auto base_iter = base_files.begin();
+      auto base_end = base_files.end();
+      auto added_iter = added_files.begin();
+      auto added_end = added_files.end();
+      while (added_iter != added_end || base_iter != base_end) {
 #ifndef NDEBUG
-        if (level > 0 && prev_file != nullptr) {
+        if (level > 0 && prev_file != nullptr && added_iter != added_end) {
           assert(base_vstorage_->InternalComparator()->Compare(
-                     prev_file->smallest, added->smallest) <= 0);
+                     prev_file->smallest, (*added_iter)->smallest) <= 0);
+        }
+        if (added_iter != added_end) {
```

Comment:
A better name for `prev_file` might be `prev_added_file`

Reply:
@ajkr I've updated the variable name. The check logic of `NDEBUG` part for sorted `added_files` moving out merge process is more clear and straightforward.